wordpress-mobile/WordPress-iOS
# Extract duplication from ReaderStreamViewController and ReaderSavedPostsViewController

*Please, so not review until PR #9384 has been merged into `develop`. This PR starts where #9384 ends*

Implements #9351 

- Implement card actions (Save for later, like, share, etc) in a set of classes, one per card action.
- Integrate this card actions into `ReaderStreamViewController` and `ReaderSavedPostsViewController`.
- Take the opportunity to remove duplication from both view controllers. 

![img_ec34df72b7f0-1](https://user-images.githubusercontent.com/2722505/40156173-a1b92c52-594c-11e8-912b-87448e2642f2.jpeg)

I think it would still be posible to extract more boilerplate from the `ReaderStreamViewController`, but in my opinion that would be beyond the scope of the current task (which is implementing Save for Later in the cleanest possible way)

To test:
1. Activate `saveForLater` feature flag
2. Navigate to `ReaderStreamViewController`. Test that post actions still work 
3. Navigate to `ReaderSavedPostsViewController`. Test that post actions work

Next steps (That I am aware of):
- Implement no results view in `ReaderSavedPostsViewController`
- Implement navigation to "View All" in the Notice presented when a post is marked as Saved for later
